core/API/ResponseBuilder.php

```diff
@@ -189,6 +189,23 @@ private function handleArray($array)
 
         $isAssoc = !empty($firstArray) && is_numeric($firstKey) && is_array($firstArray) && count(array_filter(array_keys($firstArray), 'is_string'));
 
+        if (is_numeric($firstKey)) {
+            $columns = Common::getRequestVar('filter_column', false, 'array', $this->request);
+            $pattern = Common::getRequestVar('filter_pattern', '', 'string', $this->request);
+
+            if ($columns != array(false) && $pattern !== '') {
+                $pattern = new Pattern(new DataTable(), $columns, $pattern);
+                $array   = $pattern->filterArray($array);
+            }
+
+            $limit  = Common::getRequestVar('filter_limit', -1, 'integer', $this->request);
+            $offset = Common::getRequestVar('filter_offset', '0', 'integer', $this->request);
+
+            if ($this->shouldApplyLimitOnArray($limit, $offset)) {
+                $array = array_slice($array, $offset, $limit, $firstKey !== 0);
```

Comment:
`filter_limit` was not applied if the array had numeric indexes and contained associative arrays. This does now work. 

Problem: In `plugins/APIAPI/Controller.php` we do apply a default limit if no `filter_limit` is set. To be backwards compatible we cannot apply this default filter limit on such API responses. Otherwise eg `SitesManager.getSitesWithAtLeastViewAccess` would suddenly only return 100 sites instead of all as it was before. More about this further below